### DISCUSSION TITLE
Same email can register multiple times if email contains modifiers

### DISCUSSION THREAD

**Discussion by alexferrari88** [fastapi-users/fastapi-users#599] [Ideas]
If I have a user already registered with king.arthur@camelot.bt, I can register another user with the same email but with a modifier king.arthur+fake@camelot.bt
Even though the emails look different, they are actually the same email address.

**Comment by frankie567** [comment]
Hi @alexferrari88!

Actually, I don't agree that we should ignore email aliases by default. I've tested on several services (GitHub and Twitter in particular), and they accept to create a new account with an email alias. I think it's a valid use-case generally speaking.

Now, if you want to prevent it in your own project, I suggest that you port your fix into a subclass of the database adapter. It'll allow you to reuse the routers transparently. Example with Tortoise:

```py
class CustomTortoiseUserDatabase(TortoiseUserDatabase):
    async def get_by_email(self, email: str):
        no_plus_email = re.sub(r"(\+.*)@", "@", user.email)
        return super().get_by_email(no_plus_email)
```

**Comment by alexferrari88** [comment]
Hi @frankie567 , thanks for your answer.
In my case, I want to create a subscription product which has a free tier.
If I allow people to register with modified email addresses, that's sort of a hack that someone can use to "game" the system.
I guess, allowing this or not really depends on the use case.

Thank you for the suggestion, I'll implement that in my project!

**Comment by bfontaine** [comment]
> Even though the emails look different, they are actually the same email address.

No they aren’t. This `+` modifier is a feature of Gmail only; it’s not part of the standard specification of email addresses.

**Comment by alexferrari88** [comment]
good to know, thank you @bfontaine
